src/System.Slices/System/SpanExtensions_binary.cs

```diff
@@ -58,7 +58,37 @@ public static partial class SpanExtensionsLabs
             where T : struct
         {
             Contract.RequiresInInclusiveRange(Unsafe.SizeOf<T>(), (uint)slice.Length);
-            return slice.Cast<byte, T>()[0];
+            return Unsafe.As<byte, T>(ref slice.DangerousGetPinnableReference());
+        }
+
+        /// <summary>
+        /// Reads a structure of type T out of a slice of bytes.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryRead<[Primitive]T>(this ReadOnlySpan<byte> slice, out T value)
+            where T : struct
+        {
+            if (Unsafe.SizeOf<T>() > (uint)slice.Length) {
```

Comment:
We have a discussion on this. There does not seem to be consensus that we want that. Once we have concensus, there will be thousands of places to change.